### ISSUE TITLE
Syntax highlighting of ternary if statements get confused with object creation and strings

### ISSUE THREAD

**Issue by triforce930** [atom/language-java#100]
### Prerequisites
* [X] Done, none of them, except searching for an already filed bug report. They seemed irrelevant for this issue.
* [ ] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Syntax highlighting for Java ternary operator is confused when used with the 'new' keyword and strings that have ':' character in them.
```((1==1) ? new Date().toString() + " : " : "");```
The above example makes the first '"' character not start a string. Therefore '?' matches with the first ':' it sees. It also breaks the syntax highlighting for the rest of the file. Although the problem can be easily solved by getting 'new Date()' into paranthesis, this is still a valid Java code and should be highlighted correctly.

### Steps to Reproduce

1. Set grammar to Java
2. type this: 
```((1==1) ? new Date().toString() + " : " : "");```
3. Type more lines below to see the effect.

**Expected behavior:** '?' should only match the first valid ':' character, which is the second one in my case.

**Actual behavior:** It matches the first one it sees instead.

**Reproduces how often:** Always

### Versions

OS: macOS: Sierra
```
$ atom --version
Atom    : 1.18.0
Electron: 1.3.15
Chrome  : 52.0.2743.82
Node    : 6.5.0
$ apm --version
apm  1.18.1
npm  3.10.10
node 6.9.5 x64
python 2.7.13
git 2.11.0
```

**Comment by sadikovi** [comment]
Thanks for reporting this! It looks like it is one of those problems with `new` usages in different statements. I tried solving one in arrays in my pull request #98, but it looks like we need something more general to solve this problem. Will try fixing it this week.

**Comment by 50Wliu** [comment]
Fixed by #98.
